pkg/artifactory/datasource/repository/federated/datasource_artifactory_federated_alpine_repository.go

```diff
@@ -0,0 +1,61 @@
+package federated
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/jfrog/terraform-provider-artifactory/v7/pkg/artifactory/datasource/repository"
+	resource_repository "github.com/jfrog/terraform-provider-artifactory/v7/pkg/artifactory/resource/repository"
+	"github.com/jfrog/terraform-provider-artifactory/v7/pkg/artifactory/resource/repository/federated"
+	"github.com/jfrog/terraform-provider-artifactory/v7/pkg/artifactory/resource/repository/local"
+	"github.com/jfrog/terraform-provider-shared/packer"
+	"github.com/jfrog/terraform-provider-shared/predicate"
+	"github.com/jfrog/terraform-provider-shared/util"
+)
+
+func DataSourceArtifactoryFederatedAlpineRepository() *schema.Resource {
+	packageType := "alpine"
+
+	type AlpineFederatedRepositoryParams struct {
```

Comment:
Why are we redefining the struct and schema again? They already exist in the federated repository resource.

Reply:
The schema needs to redefined here because MemberSchema has been redefined. 

However, we can reuse the structs, will change that now.